Cpp_hands_on_programming/lib/pipe/Ipc_pipe.h

```diff
@@ -0,0 +1,16 @@
+#ifndef IPC_PIPE
+#define IPC_PIPE
+#include "lib/abstract/Ipc_method.h"
+
+class Ipc_pipe : public Ipc_method{
+private:
+    std::string name;
+    std::string filename;
+public:
+    Ipc_pipe(std::string name0 = "", std::string file0 = "") : name(name0), filename(file0) {}
```

Comment:
Is there any use of default arguments here? you can't set name and filename later

Reply:
I set it for the sake of creating a 2 in 1 initializer list constructor so I don't need one default constructor and one constructor with 2 parameters. Is this a bad practice for some reason?

Reply:
Hmm, I actually tried that out with default constructor, e,g Ipc_info info, that seems to work. Do you refer to something else? 

As for the empty string value of the member variable, would that be the same thing if I create a separate default constructor without initializing those member variables, i.e empty string?

Reply:
Sorry, maybe there was misunderstanding :)
Why you even want this 'default constructor behavior'?  For my understanding there is no usage of something like: Ipc_info info , or is it?

Reply:
I see what you mean. My example with Ipc_info is a bad one, the default constructor behavior is used in the ipc_sendfiles and ipc_receivefiles files, because Ipc_info has setters. 

You are right about Ipc_pipe, it does not need default constructor behavior. I forgot to consider that when I copy paste the constructor from Ipc_info to Ipc_pipe. Thx!